cvl/cvl.mjs

```diff
@@ -6,213 +6,263 @@ import cvlParser from './.antlr/cvlParser.mjs';
+		return Number(ctx.getText());
 	}
 
 
 	// Visit a parse tree produced by cvlParser#longNumberLiteral.
 	visitLongNumberLiteral(ctx) {
-    return BigInt(ctx.getText().slice(0, -1));
+		return BigInt(ctx.getText().slice(0, -1));
 	}
 
 
 	// Visit a parse tree produced by cvlParser#dateTimeLiteral.
 	visitDateTimeLiteral(ctx) {
-	  //return this.visitChildren(ctx);
-    return ctx.getText();
+		//return this.visitChildren(ctx);
+		return ctx.getText();
 	}
 
 
 	// Visit a parse tree produced by cvlParser#dateLiteral.
 	visitDateLiteral(ctx) {
-	  //return this.visitChildren(ctx);
-    return ctx.getText();
+		//return this.visitChildren(ctx);
+		return ctx.getText();
 	}
 
 
 	// Visit a parse tree produced by cvlParser#timeLiteral.
 	visitTimeLiteral(ctx) {
-	  //return this.visitChildren(ctx);
-    return ctx.getText();
+		//return this.visitChildren(ctx);
+		return ctx.getText();
 	}
 
 
 	// Visit a parse tree produced by cvlParser#quantityLiteral.
 	visitQuantityLiteral(ctx) {
-	  //return this.visitChildren(ctx);
-    return ctx.children[0].accept(this);
+		//return this.visitChildren(ctx);
+		return ctx.children[0].accept(this);
 	}
 
 
 	// Visit a parse tree produced by cvlParser#ratioLiteral.
 	visitRatioLiteral(ctx) {
-	  //return this.visitChildren(ctx);
-    return ctx.children[0].accept(this);
+		//return this.visitChildren(ctx);
+		return ctx.children[0].accept(this);
 	}
 
 
 	// Visit a parse tree produced by cvlParser#intervalSelector.
 	visitIntervalSelector(ctx) {
-	  return this.visitChildren(ctx);
+		return {
+			lowClosed: ctx.children[1].accept(this) === '[',
+			low: ctx.children[2].accept(this),
+			hiClosed: ctx.children[5].accept(this) === ']',
+			hi: ctx.children[4].accept(this)
+		}
 	}
 
 
 	// Visit a parse tree produced by cvlParser#tupleSelector.
 	visitTupleSelector(ctx) {
-	  return this.visitChildren(ctx);
+		var result = { };
+		for (let i = 0; i < ctx.children.length; i++) {
+			var child = ctx.children[i].accept(this);
+			if (child === 'Tuple') {
+				continue;
+			}
+
+			if (child === ':') {
+				continue;
+			}
+
+			if (child === '{') {
+				continue;
+			}
+
+			if (child === '}') {
+				continue;
+			}
+
+			Object.assign(result, child);
+		}
+		return result;
 	}
 
 
 	// Visit a parse tree produced by cvlParser#tupleElementSelector.
 	visitTupleElementSelector(ctx) {
-	  return this.visitChildren(ctx);
+		return { [ctx.children[0].accept(this)]: ctx.children[2].accept(this) };
 	}
 
 
 	// Visit a parse tree produced by cvlParser#instanceSelector.
 	visitInstanceSelector(ctx) {
-	  return this.visitChildren(ctx);
+		var result = { };
+		for (let i = 0; i < ctx.children.length; i++) {
+			var child = ctx.children[i].accept(this);
+			if (i === 0) {
+				continue;
+			}
+
+			if (child === ':') {
+				continue;
+			}
+
+			if (child === '{') {
+				continue;
+			}
+
+			if (child === '}') {
+				continue;
+			}
+
+			Object.assign(result, child);
+		}
+		return result;
```

Comment:
For this visitor behavior in general (cvlParseVisitor), is there somewhere we can add a few sanity unit tests for various inputs? WDYT? We don't need to block on it if you feel it doesn't make sense at this stage and will be covered by downstream test runner tests.

Reply:
Can we add unit tests as another step? I'd be happy to add them but I want to try to keep things moving and it will be part of the next step (debugging the rest of the failing tests) to make sure this part is working as expected. I'll submit an issue to add unit tests for the cvlVisitor